variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/AbstractVariant.java

```diff
@@ -102,7 +102,7 @@ protected AbstractVariant() {
         this.hgvs = new HashMap<>();
     }
 
-    public AbstractVariant(String chromosome, long start, long end, String reference, String alternate) {
+    public AbstractVariant(String chromosome, long start, long end, String reference, String alternate, String mainId) {
```

Comment:
I wonder if we should have 2 constructors, one with and another without the `mainId` parameter. The one without it will be used by the pipeline that creates new variants, ensuring that we don't assign the ID even by mistake. The one with it will be used by the web services. We could document them to make clear when each of them should be used.

Reply:
Agree